migrations/20221212085251-fix-expense-refunds-with-payment-processor-cover.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  async up(queryInterface) {
+    await queryInterface.sequelize.query(`WITH "ExpenseRefundsCreditsTransactionsUpdate" AS (
+      SELECT tr.id as "id",
+        - (t."amount" ) as "amount",
+        - (t."amountInHostCurrency") as "amountInHostCurrency",
+        - (t."netAmountInCollectiveCurrency" - t."paymentProcessorFeeInHostCurrency") as "netAmountInCollectiveCurrency"
+      FROM "Transactions" t
+      INNER JOIN "Transactions" tr ON t."RefundTransactionId" = tr."id" AND tr."deletedAt" IS NULL
+      WHERE t."kind" = 'EXPENSE'
+        AND t."RefundTransactionId" IS NOT NULL
+        AND t."type" = 'DEBIT'
+        AND t."paymentProcessorFeeInHostCurrency" != 0
+        AND t."currency" = t."hostCurrency"
+        AND t."deletedAt" IS NULL
+        AND EXISTS (SELECT * FROM "Transactions" WHERE "kind" = 'PAYMENT_PROCESSOR_COVER' AND "TransactionGroup" = tr."TransactionGroup")
+      )
+      UPDATE "Transactions"
+      SET "amount" = erctu."amount",
+        "amountInHostCurrency" = erctu."amountInHostCurrency",
+        "netAmountInCollectiveCurrency" = erctu."netAmountInCollectiveCurrency",
+        "updatedAt" = NOW()
+      FROM "ExpenseRefundsCreditsTransactionsUpdate" erctu
+      WHERE "Transactions"."id" = erctu."id"`);
+
+    await queryInterface.sequelize.query(`WITH "ExpenseRefundsDebitsTransactionsUpdate" AS (
+      SELECT tr.id as "id",
+        - (t."amount" + t."paymentProcessorFeeInHostCurrency") as "amount",
+        - (t."amountInHostCurrency" + t."paymentProcessorFeeInHostCurrency") as "amountInHostCurrency",
+        - (t."netAmountInCollectiveCurrency") as "netAmountInCollectiveCurrency"
+      FROM "Transactions" t
+      INNER JOIN "Transactions" tr ON t."RefundTransactionId" = tr."id" AND tr."deletedAt" IS NULL
+      WHERE t."kind" = 'EXPENSE'
+        AND t."RefundTransactionId" IS NOT NULL
+        AND t."type" = 'CREDIT'
+        AND t."paymentProcessorFeeInHostCurrency" != 0
+        AND t."currency" = t."hostCurrency"
+        AND t."deletedAt" IS NULL
+        AND EXISTS (SELECT * FROM "Transactions" WHERE "kind" = 'PAYMENT_PROCESSOR_COVER' AND "TransactionGroup" = tr."TransactionGroup")
+      )
+      UPDATE "Transactions"
+      SET "amount" = erdtu."amount",
+        "amountInHostCurrency" = erdtu."amountInHostCurrency",
+        "netAmountInCollectiveCurrency" = erdtu."netAmountInCollectiveCurrency"
+      FROM "ExpenseRefundsDebitsTransactionsUpdate" erdtu
+      WHERE "Transactions"."id" = erdtu."id"`);
+  },
+
+  async down() {
+    // No rollback
+  },
+};
```

Comment:
The migration seems correct, I only noticed we have a few transactions with an unusually high payment processor fee, for example, 1744368, 3431872, 2906770.
What is this about?

Reply:
I guess people messed up when "Marking as paid" and entered the expense amount instead of the payment processor fees.

<img width="358" alt="Screenshot 2022-12-12 at 15 39 56" src="https://user-images.githubusercontent.com/806/207073320-92b21ba5-c50c-4e9d-81c8-ff5ef5c4fdc9.png">